File: programmers/MakeSumEqual.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
  public int solution(int[] queue1, int[] queue2) {
    long sum1 = 0;
    long sum2 = 0;
    int midVal = 0;
    Queue<Long> first = new LinkedList<Long>();
    Queue<Long> second = new LinkedList<Long>();
    int cnt = 0;
    for (int i = 0; i < queue1.length; i++) {
      sum1 += queue1[i];
      first.offer((long) queue1[i]);
    }
    for (int i = 0; i < queue2.length; i++) {
      sum2 += queue2[i];
      second.offer((long) queue2[i]);
    }

    if ((sum1 + sum2) % 2 != 0)
      return -1;

    while (true) {
      if (sum1 > sum2) {
        long tem = first.poll();
        sum1 -= tem;
        sum2 += tem;
        second.offer(tem);
        cnt++;
      } else if (sum1 < sum2) {
        long tem = second.poll();
        sum2 -= tem;
        sum1 += tem;
        first.offer(tem);
        cnt++;
      } else if (sum1 == sum2) {
        System.out.println(cnt);
        return cnt;
      }

      if (cnt > (queue1.length + queue2.length) * 2)
        return -1;
    }
  }
}

File: programmers/MakeSumEqual.py
from queue import Queue

def solution(queue1, queue2):
  que1 = Queue()
  que2 = Queue()
  
  sum1 = 0
  sum2 = 0
  for i in range(len(queue1)):
    sum1 += queue1[i]
    que1.put(queue1[i])
  for i in range(len(queue2)):
    sum2 += queue2[i]
    que2.put(queue2[i])
    
  if((sum1 + sum2) % 2) != 0:
    return -1
  
  cnt = 0
  while(True):
    if sum1 > sum2:
      tem = que1.get()
      sum1 -= tem
      sum2 += tem
      que2.put(tem)
      cnt += 1
    elif sum1 <sum2:
      tem = que2.get()
      sum1 += tem
      sum2 -= tem
      que1.put(tem)
      cnt += 1
    elif sum1 == sum2:
      return cnt
    if cnt > (len(queue1)  + len(queue2)) * 2: 
      return -1